src/main/java/de/tum/in/www1/artemis/repository/ProgrammingSubmissionRepository.java

```diff
@@ -52,15 +52,37 @@ List<ProgrammingSubmission> findByParticipationIdAndCommitHashOrderByIdDescWithF
     default ProgrammingSubmission findFirstByParticipationIdAndCommitHashOrderByIdDescWithFeedbacksAndTeamStudents(Long participationId, String commitHash) {
         return findByParticipationIdAndCommitHashOrderByIdDescWithFeedbacksAndTeamStudents(participationId, commitHash).stream().findFirst().orElse(null);
     }
+    /*
+     * @Query("""
+     * SELECT s
+     * FROM ProgrammingSubmission s
+     * LEFT JOIN FETCH s.results
+     * WHERE (s.type <> de.tum.in.www1.artemis.domain.enumeration.SubmissionType.ILLEGAL OR s.type IS NULL)
+     * AND s.participation.id = :participationId
+     * AND s.id = (
+     * SELECT MAX(s2.id)
+     * FROM ProgrammingSubmission s2
+     * WHERE s2.participation.id = :participationId
+     * AND (s2.type <> de.tum.in.www1.artemis.domain.enumeration.SubmissionType.ILLEGAL OR s2.type IS NULL)
+     * )
+     * """)
+     */
 
     @Query("""
-            SELECT s FROM ProgrammingSubmission s
-            LEFT JOIN FETCH s.results
+            SELECT s
+            FROM ProgrammingSubmission s
+                LEFT JOIN FETCH s.results
             WHERE (s.type <> 'ILLEGAL' or s.type is null)
-            AND s.participation.id = :#{#participationId}
-            AND s.id = (SELECT max(s2.id) FROM ProgrammingSubmission s2 WHERE s2.participation.id = :#{#participationId} AND (s2.type <> 'ILLEGAL' or s2.type is null))
+                AND s.participation.id = :participationId
+                AND s.id = (
+                    SELECT MAX(s2.id)
```

Comment:
we want to avoid sub queries if possible, because they can lead to bad performance

Reply:
Yes, I'm trying to understand if I could refactor this one; I think the error for the failing tests lies herein.